### DISCUSSION TITLE
Do manifest applications have to be a git repo?

### DISCUSSION THREAD

**Discussion by kieranwlc** [zephyrproject-rtos/zephyr#66064] [Q&A] [answered]
Apologies if this has been covered, I have searched but couldn't find this exact discussion with the terms i was using.

Okay so I've currently got a repository that looks like this:

Project Repository/ - git repo
├── A
├── B
├── West Workspace
│   └── Foo Application (containing west.yml file)

I want to be able to fetch this and run west init, targeting the west.yml file, and initialising the west workspace in /West Workspace with all the relevant modules etc.

I've read some of the docs, so am I correct in concluding that in order to do this, I am required to make Foo Application into it's own repository, and set it up as a git sub-module of my Project Repository?

If so... why?

**Comment by cameled** [comment]
After `west init`, `west` will create a file under `West Workspace`, named `.west/config`,  contain below config.
```
[manifest]
path = Foo_Application
file = west.yml
```

There no need to run 'west init' again, if you already have this file. So, this may help you to understand what `west init` done.

**Comment by blemouzy** [comment, marked as answer]
No, you can do something like that:

```
west init -l --mf west.yml "West Workspace/Foo Application"
```

or

```
cd "West Workspace"
west init -l --mf west.yml "Foo Application"
```
